appcli/models/configuration.py

```diff
@@ -112,6 +120,21 @@ class Configuration(NamedTuple):
     generated configuration directory.
     """
 
+    def get_app_name_shell_safe(self) -> str:
+        """A shell safe version of the application name.
+        This transforms the app_name variable by replacing any unsafe shell
+        characters with '_', and returning the new string.
+
+        Returns:
+            The app_name with no unsafe shell characters (echo-server -> echo_server) etc.
+            Or a custom shell-safe name provided by the user.
+
+        """
+        if self.app_name_shell_safe is None:
+            # User hasn't defined a custom shell-safe name.
+            return re.sub(r"[\s\-\.\{\}\[\]\"\'\,]", "_", self.app_name)
```

Comment:
Is the the whole set of unsafe shell characters? Defining the full set of unsafe characters is going to be difficult, as there's almost always going to be cases that we miss.

Reply:
This is just a list of the characters that I thought may be a problem.
I could define the characters as a set, that we could add to as we find more.
Or I could have a look around the internet to see if anyone has anything that we could pull from, that is more of a complete list.

Reply:
See https://linuxhint.com/bash-variable-name-rules-legal-illegal/

> A variable name can have letter/s.
> A variable name can have numbers, underscores, and digits.

Reply:
The current regex I have is: `[^a-zA-Z0-9_]` which seems to cover that. I am still working on a better way to do the getter logic though.

Reply:
There's more detail in this answer here too: https://unix.stackexchange.com/questions/428880/list-of-acceptable-initial-characters-for-a-bash-variable
I don't think it can start with a number, so you may need to consider that case too

Reply:
I've changed the regex to (hopefully) better meet bash character limitations, and removed the 'if' logic with exception handling.